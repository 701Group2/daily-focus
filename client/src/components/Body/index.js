import SelectedWidgetsModal from "../SelectWidgetsModal";
import styles from "./style.module.scss";

import CalendarWidget from "../CalendarWidget";
import Timer from "../Timer";
import ToDoList from "../ToDoList";

import NewsWidget from "../NewsWidget";
import FactOfTheDay from "../FactOfTheDay";
import SpotifyWidget from "../SpotifyWidget";


function Body({ selectedWidgets, setSelectedWidgets }) {
    return (
        <div className={styles.container}>
            {selectedWidgets && selectedWidgets.length ? (
                <div className={styles.grid}>
                    {selectedWidgets.includes("To Do List") && (
                        <div className={styles.widget}>
                            <ToDoList />
                        </div>
                    )}
                    {selectedWidgets.includes("Calendar") && (
                        <div className={styles.widget}>
                            <CalendarWidget />
                        </div>
                    )}
                    {selectedWidgets.includes("Timer") && (
                        <div className={styles.widget}>
                            <Timer />
                        </div>
                    )}
                    {selectedWidgets.includes("News Widget") && (
                        <div className={styles.widget}>
                            <NewsWidget />
                        </div>
                    )}
                    {selectedWidgets.includes("Fact Of The Day") && (
                        <div className={styles.widget}>
                            <FactOfTheDay />
                        </div>
                    )}
                    {selectedWidgets.includes("Spotify") && (
                        <div className={styles.widget}>
                            <SpotifyWidget />
                        </div>
                    )}
                </div>
            ) : (
                <div className={styles.noWidgetContainer}>
                    <div className={styles.message}>Your focus page is empty</div>
                    <div className={styles.addWidget}>
                        <SelectedWidgetsModal
                            selectedWidgets={selectedWidgets}
                            setSelectedWidgets={setSelectedWidgets}
                        />
                    </div>
                </div>
            )}
        </div>
    );
}

export default Body;
