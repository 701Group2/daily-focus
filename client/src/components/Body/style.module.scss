.gridContainer {
    height: 100%;
    background-color: #f0f2f5;
}

.noWidgetContainer {
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    margin-right: -50%;
    transform: translate(-50%, -50%);
    text-align: center;

    .message {
        font-size: 1.5rem;
        color: #7d7d7d;
    }

    .addWidget {
        margin-top: 15px;
        text-align: center;
    }
}
