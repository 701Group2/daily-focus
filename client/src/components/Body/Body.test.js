import React from "react";
import { shallow } from "enzyme";
import Body from ".";
import renderer from "react-test-renderer";

test("Renders body without crashing", () => {
    shallow(<Body />);
});

test("Renders no widgets found message", () => {
    const tree = renderer.create(<Body />).toJSON();

    expect(tree).toMatchSnapshot();
});
