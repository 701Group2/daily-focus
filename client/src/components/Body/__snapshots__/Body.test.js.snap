// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders no widgets found message 1`] = `
<div
  className="container"
>
  <div
    className="noWidgetContainer"
  >
    <div
      className="message"
    >
      Your focus page is empty
    </div>
    <div
      className="addWidget"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Manage Widgets
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
